controllers/autoscaling.go

```diff
@@ -52,18 +52,37 @@ func (r *HorizontalRunnerAutoscalerReconciler) determineDesiredReplicas(rd v1alp
 			return nil, err
 		}
 
-		for _, r := range list.WorkflowRuns {
+		for _, run := range list.WorkflowRuns {
 			total++
 
 			// In May 2020, there are only 3 statuses.
 			// Follow the below links for more details:
 			// - https://developer.github.com/v3/actions/workflow-runs/#list-repository-workflow-runs
 			// - https://developer.github.com/v3/checks/runs/#create-a-check-run
-			switch r.GetStatus() {
+			switch run.GetStatus() {
 			case "completed":
 				completed++
 			case "in_progress":
-				inProgress++
+				jobs, _, err := r.GitHubClient.Actions.ListWorkflowJobs(context.TODO(), user, repoName, run.GetID(), nil)
```

Comment:
In general the change looks good to me and should be fine as long as we don't monitor too many repos with many jobs, otherwise we would indeed likely hit the API rate limit fast.

But yeah, I checked the Github API and there is not much more we can do when utilizing the polling approach.

One additional thing / improvement we could do is utilize ``If-Modified-Since`` header (https://docs.github.com/en/free-pro-team@latest/rest/overview/resources-in-the-rest-api#conditional-requests).

This way request would not count against rate limit quota if nothing has changed, but then again I don't think we would hit that scenario all that often in this list workflows and list workflow jobs scenario which is pretty dynamic / changing in nature (although it also kinds depends on the polling interval).

Reply:
I had a quick look at the go client - https://github.com/google/go-github/blob/master/github/actions_workflows.go#L65

It looks like we would need to modify the client to support passing custom headers to utilize If-Modified-Since functionality  (which may be a good upstream contribution anyway).

Reply:
Good call on the `If-Modified-Since` header, I'll add that!

I plan to iterate on this and eventually add a webhook and just use those events to scale.